### DISCUSSION TITLE
Nested Model in request body producing only default values

### DISCUSSION THREAD

**Discussion by david-saeger** [fastapi/fastapi#6488] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from pydantic import BaseModel, Field
from typing import List, Optional
router = APIRouter()
from fastapi import APIRouter, Depends

class Contract(BaseModel):
    contract_uuid: str = Field(
        None,
        alias="CONTRACT_UUID",
,
    )
    contract_id: Optional[str] = Field(
        None,
        alias="CONTRACT_ID",

    )
    contract_name: Optional[str] = Field(
        None,
        alias="CONTRACT_NAME",
        title="Project/Contract Name",

    )
   
    full_part_time: Optional[EmploymentType] = Field(
        None,
        alias="FULL_PART_TIME",
        title="Employment Type",
 
    )


class UpdateContracts(BaseModel):
    contract_array: List[Contract] = Field(
        [], description="A list of contracts submitted by the employer"
    )
    case_id: str = Field(
        None,
        title="Case ID",
      
    )
    company_id: str = Field(
        None,
        title="Company Identifier",

    )


@router.put("/contracts/")
async def put_contracts(data: UpdateContracts):
    """ Update contracts for one applicant """
    print(data.json())
```

### Description

I am building an API that  returns a nested model UpdateContracts which contains some fields and a list of contracts. The fields show up fine but in the nested list of Contracts I can only produce the default null values specified in the Contract model. I can validate that the appropriately shaped data is being sent, here is the payload (it contains some extrainious fields but that should not matter): 
```
from fastapi import APIRouter, Body, Depends
{"company_id":"pae","nvc_siv_case_id":"Ip0XkAZc","contract_array":[{"contract_id":"flomo","contract_name":"asdf","badge_id":"12345","job_location":null,"full_part_time":"FULLTIME","reason_for_separation":"currently_employed","employment_details":null,"employment_details_other":null,"contract_uuid":"909a1d12-43fe-42d9-ba0d-56c74aa167d2"}]}

```

### Operating System

macOS

### FastAPI Version

fastapi==0.61.1

### Python Version

3.8

**Comment by harunyasar** [comment, marked as answer]
I have to say I felt like I was lost in the first place. But I believe you should use aliases instead of the field definitions. What I mean by saying that you can see as follows.

```python
import inspect
inspect.signature(Contract)
# <Signature (*, CONTRACT_UUID: str = None, CONTRACT_ID: str = None, CONTRACT_NAME: str = None) -> None>
```

Which means your `Contract` object's signature should have been like above one. So,

```python
update_contract = UpdateContracts(
    case_id="1230",
    company_id="3210",
    contract_array=[
        Contract(
            CONTRACT_UUID="c4469310-b22e-4305-a530-3b8b8dc15b6f",
            CONTRACT_ID="9876",
            CONTRACT_NAME="Example contract name"
        )
    ]
)

# Output
# UpdateContracts(
#     contract_array=[
#         Contract(
#             contract_uuid='c4469310-b22e-4305-a530-3b8b8dc15b6f',
#             contract_id='9876',
#             contract_name='Example contract name'
#         )
#     ], 
#     case_id='1230', 
#     company_id='3210'
# )
```

I hope that it can help.

**Comment by david-saeger** [comment]
that is helpful, is there a way for me to specify to use field definitions and not aliases?

**Comment by harunyasar** [comment]
> that is helpful, is there a way for me to specify to use field definitions and not aliases?

If you don’t want to use the aliases at all you could remove them from the field definitions.

**Comment by david-saeger** [comment]
thank you @harunyasar
